app/_kong_plugins/websocket-size-limit/index.md

```diff
@@ -33,6 +33,87 @@ categories:
 
 search_aliases:
   - websocket-size-limit
+
+tags:
+  - traffic-control
+  - websocket
+
+related_resources:
+  - text: WebSocker Validator plugin
+    url: /plugins/websocket-validator/
+  - text: WebSocket RFC
+    url: https://datatracker.ietf.org/doc/html/rfc6455
 ---
 
-## Overview
+Allows operators to specify a maximum size for incoming WebSocket messages.
+
+Separate limits can be applied to clients and upstreams.
+
+When an incoming message exceeds the limit:
+1. A close frame with status code `1009` is sent to the sender
+2. A close frame with status code `1001` is sent to the peer
+3. Both sides of the connection are closed
+
+## How it works
+
+Size limits can be applied to client messages, upstream messages, or both.
+Limits are evaluated based on the message payload length, and not the entire length of the WebSocket frame (header and payload).
+
+### Default limits
+
+{{site.base_gateway}} applies the following default limits to incoming messages for all WebSocket services:
+
+| Sender   | Default Limit        |
+|----------|----------------------|
+| client   | `1048576` (`1MiB`)   |
+| upstream | `16777216` (`16MiB`) |
+
+This plugin can be used to increase the limit beyond the default using the [`config.client_max_payload`](/plugins/websocket-size-limit/reference/#schema--config-client-max-payload) and 
+[`config.upstream_max_payload`](/plugins/websocket-size-limit/reference/#schema--config-upstream-max-payload) parameters.
+
+The default client limit is smaller than the default upstream limit because proxying client-originated messages is much more computationally expensive than upstream messages.
+This is due to the client-to-server masking required by the [WebSocket specification](https://datatracker.ietf.org/doc/html/rfc6455#section-5.3), 
+so it's generally better to maintain a lower limit for client messages.
+
+### Standalone data frames (text and binary)
+
+For limits of 125 bytes or less, the message length is checked after reading and decoding the entire message into memory.
+
+For limits of 125 bytes or more, the message length is checked from the frame header _before_ the entire message is read from the socket buffer,
+allowing {{site.base_gateway}} to close the connection without having to read, and potentially unmask, the entire message into memory.
```

Comment:
This is unclear, which option applies if the limit is exactly 125 bytes?

Reply:
Nice catch - had to look up the websocket rfc to verify this one. It's 125 or less and 126 or more.